doc/sphinx/source/howto/update.rst

```diff
@@ -0,0 +1,48 @@
+.. update.rst
+
+###########################
+Make a Stateful Computation
+###########################
+
+In this section, we show how to make a stateful computation from
+nGraphs stateless operations. The basic idea is that any computation
+with side-effects can be factored into a stateless function that
+transforms the old state into the new state.
+
+An Example from C++
+===================
+
+Let's start with a simple C++ example, a function ``count`` that
+returns how many times it has already been called:
+
+.. literalinclude:: ../../../examples/update.cpp
+   :language: cpp
+   :lines: 20-24
+
+This function has state kept in the static variable ``counter``. The
```

Comment:
This intro might need reworded or edited.  
Could this part 
> this function has state kept in the static variable ``counter``

Be reworded for clarity to 

> A static variable named ``counter`` holds the state of this function (... which is initialized to zero) 

?